### ISSUE TITLE
AWSDataStorePlugin fail to initialize and app crashes

### ISSUE THREAD

**Issue by btdo** [aws-amplify/amplify-android#563]
Hello,

I have a demo android app using Amplify Data store and once a while when I boot up the app it crashes and I would see an error in the log that seems to be related to the plugin.  If I start the app again then it would be fine.  

Here's is how I initialize the plugin:

```
 implementation 'com.amplifyframework:core:1.0.0'
 implementation 'com.amplifyframework:aws-api:1.0.0' // If using cloud sync
 implementation 'com.amplifyframework:aws-datastore:1.0.0'
```

```
try {
    val authProviders = ApiAuthProviders.builder()
        .oidcAuthProvider(MyOicAuthServiceImpl)
        .build()
    Amplify.addPlugin(AWSDataStorePlugin())
    Amplify.addPlugin(AWSApiPlugin(authProviders)) // If using remote model synchronization
    Amplify.configure(applicationContext)
    Log.i("MyAmplifyApp", "Initialized Amplify")
} catch (error: AmplifyException) {
    Log.e("MyAmplifyApp", "Could not initialize Amplify", error)
}
```

Here's the stack trace 

```
2020-06-05 16:41:33.786 8343-8407/com.wwm.trackappsyncapp E/AndroidRuntime: FATAL EXCEPTION: pool-2-thread-2
    Process: com.wwm.trackappsyncapp, PID: 8343
    java.lang.RuntimeException: java.util.concurrent.TimeoutException: The source did not signal an event for 5000 milliseconds and has been terminated.
        at io.reactivex.internal.util.ExceptionHelper.wrapOrThrow(ExceptionHelper.java:46)
        at io.reactivex.internal.observers.BlockingMultiObserver.blockingGetError(BlockingMultiObserver.java:153)
        at io.reactivex.Completable.blockingGet(Completable.java:1302)
        at com.amplifyframework.datastore.AWSDataStorePlugin.initialize(AWSDataStorePlugin.java:7)
        at com.amplifyframework.core.category.Category.initialize(Category.java:15)
        at com.amplifyframework.core.Amplify.lambda$beginInitialization$0(Amplify.java:1)
        at com.amplifyframework.core.-$$Lambda$Amplify$EB8h7rBH19uqUcPr0ZtYS0ww0r4.run(Unknown Source:4)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
```

Let me know if you need more info.

Thanks

**Comment by jamesonwilliams** [comment]
Hm, I don't think this _ough to_ matter, but I have never tried adding the DataStore plugin _before_ the API plugin. Does anything change, if you swap those two lines?

**Comment by raphkim** [comment]
@jamesonwilliams I can confirm that the ordering of adding the plugins will not matter. The sample code in documentation also adds the plugins in the above order.

**Comment by btdo** [comment]
Hi @jamesonwilliams, it didn't help, I swapped and I still see the issue.  Although, I'm not sure if this helps but I haven't seen this issue in the real phone.  The stack trace that I got was from the emulator. I'll keep trying on a real phone and see if I can recreate the issue

**Comment by marianolc** [comment]
I'm having the same issues. Note that the one causing the problem is not DataStore but Api. From the code:

```
            Amplify.addPlugin(new AWSCognitoAuthPlugin());
            Amplify.addPlugin(new AWSDataStorePlugin());
            //Amplify.addPlugin(new AWSApiPlugin()); // If using remote model synchronization
            Amplify.configure(getApplicationContext());

```

This make the app work offline. But if I remove the // in the AwsApiPlugin I get the error:

```

2020-06-16 18:00:08.916 3793-3876/com.poscloud.pos E/AndroidRuntime: FATAL EXCEPTION: pool-2-thread-3
    Process: com.poscloud.pos, PID: 3793
    java.lang.RuntimeException: java.util.concurrent.TimeoutException: The source did not signal an event for 5000 milliseconds and has been terminated.
        at io.reactivex.internal.util.ExceptionHelper.wrapOrThrow(ExceptionHelper.java:46)
        at io.reactivex.internal.observers.BlockingMultiObserver.blockingGetError(BlockingMultiObserver.java:153)
        at io.reactivex.Completable.blockingGet(Completable.java:1302)
        at com.amplifyframework.datastore.AWSDataStorePlugin.initialize(AWSDataStorePlugin.java:7)
        at com.amplifyframework.core.category.Category.initialize(Category.java:15)
        at com.amplifyframework.core.Amplify.lambda$beginInitialization$0(Amplify.java:1)
        at com.amplifyframework.core.-$$Lambda$Amplify$EB8h7rBH19uqUcPr0ZtYS0ww0r4.run(Unknown Source:4)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: java.util.concurrent.TimeoutException: The source did not signal an event for 5000 milliseconds and has been terminated.
        at io.reactivex.internal.observers.BlockingMultiObserver.blockingGetError(BlockingMultiObserver.java:153) 
        at io.reactivex.Completable.blockingGet(Completable.java:1302) 
        at com.amplifyframework.datastore.AWSDataStorePlugin.initialize(AWSDataStorePlugin.java:7) 
        at com.amplifyframework.core.category.Category.initialize(Category.java:15) 
        at com.amplifyframework.core.Amplify.lambda$beginInitialization$0(Amplify.java:1) 
        at com.amplifyframework.core.-$$Lambda$Amplify$EB8h7rBH19uqUcPr0ZtYS0ww0r4.run(Unknown Source:4) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764) 
```

**Comment by marianolc** [comment]
I will add something: Clearing app data and cache let the app to initialize without problem. But at the moment I restart the app the crash happens again.

Tried in : 

- Emulator Api 28
- Phone Redmi Note 7 

With the same issue.

**Comment by tomasfelder** [comment]
Hi there,
I'm having the same issue while using Amplify Datastore. I've upgraded my project to use Datastore because I was using the AppSync old way to communicate with the API. I needed DataStore because I need my app to work offline and then syncronize the data with the cloud. I followed all the steps to get it working but I cannot face why is it not working. My app crashes all time i run it with this expcetion. I'm using Kotlin language and Cognito Auth mode

E/AndroidRuntime: FATAL EXCEPTION: pool-2-thread-3 Process: com.example.scanningapp, PID: 2807 java.lang.RuntimeException: java.util.concurrent.TimeoutException: The source did not signal an event for 5000 milliseconds and has been terminated. at io.reactivex.internal.util.ExceptionHelper.wrapOrThrow(ExceptionHelper.java:46) at io.reactivex.internal.observers.BlockingMultiObserver.blockingGetError(BlockingMultiObserver.java:153) at io.reactivex.Completable.blockingGet(Completable.java:1302) at com.amplifyframework.datastore.AWSDataStorePlugin.initialize(AWSDataStorePlugin.java:7) at com.amplifyframework.core.category.Category.initialize(Category.java:15) at com.amplifyframework.core.Amplify.lambda$beginInitialization$0(Amplify.java:1) at com.amplifyframework.core.-$$Lambda$Amplify$EB8h7rBH19uqUcPr0ZtYS0ww0r4.run(Unknown Source:4) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636) at java.lang.Thread.run(Thread.java:764)
Thank you in advance!
